### PR TITLE
Make -j and -j0 act like on GNU Make.

### PR DESCRIPTION
**Pull request by xim**

### CODE REVIEW

File: src/ninja.cc [hunk 1]
```diff
@@ -1058,11 +1058,14 @@ int ReadFlags(int* argc, char*** argv,
...
+        if (!optarg)
+          break;
         char* end;
         int value = strtol(optarg, &end, 10);
-        if (*end != 0 || value <= 0)
+        if (*end != 0 || value < 0)
           Fatal("invalid -j parameter");
-        config->parallelism = value;
+        if (value)
+          config->parallelism = value;
```
Comment (stefanb2) [inline src/ninja.cc]: That just means that `-j` or `-j0` will be cause ninja to use the default parallelism, which is set up in line 1039 above.

I definitely doesn't mean the same as GNU make `-j` or `-j0` i.e. run as many jobs in parallel as are ready to run.

**Comment by stefanb2** [comment]
This PR has nothing to do with PR #1140 and is not required for it.

File: src/ninja.cc (hunk 1 above)
Comment (xim) [reply src/ninja.cc]: True, I could make it INT_MAX. I'll also drop the reference to PR #1140. The reason I mention it is that our build system passes a sanitized version of the `MAKEFLAGS` variable to the command line of the builders. As that by default contains a `-j` in an environment where Make <4.2 has a command called with a `+`-prefixed line, I feel they're related (although not needed to be useful individually).

**Comment by xim** [comment]
Pull request updated with INT_MAX jobs and reference to other PR dropped.

**Comment by nico** [comment]
Why?

**Comment by nico** [comment]
I'm ok with supporting `-j 0` since that's consistent with `-k 0` (you should update --help output to say "(0 means infinity)" like the help now says for -k).

I don't see the point of supporting `-j` without an argument.

**Comment by jhasse** [comment]
Thanks for the PR! As nico said, please add some documentation to `--help` and drop the support of `-j` without an argument. I'm closing this in order to reduce the number of open PRs, but with those changes we would merge this.

**Pull request closed without merging**